src/braket/circuits/pauli_string.py

```diff
@@ -0,0 +1,206 @@
+                string or another PauliString object. A valid string consists of an optional phase,
+                specified by an optional sign +/- followed by an uppercase string in {I, X, Y, Z}.
+                Example valid strings are: XYZ, +YIZY, -YX
+        """
+        if not pauli_string:
+            raise ValueError("pauli_string must not be empty")
+        if isinstance(pauli_string, PauliString):
+            self._phase = pauli_string._phase
+            self._qubit_count = pauli_string._qubit_count
+            self._nontrivial = pauli_string._nontrivial
+            self._eigenstate_circuits = pauli_string._eigenstate_circuits
+        elif isinstance(pauli_string, str):
+            self._phase, factors_str = PauliString._split(pauli_string)
+            self._qubit_count = len(factors_str)
+            self._nontrivial = {
+                i: factors_str[i] for i in range(len(factors_str)) if factors_str[i] != "I"
+            }
+            self._eigenstate_circuits = {}
+        else:
+            raise TypeError(f"Pauli word {pauli_string} must be of type {PauliString} or {str}")
+
+    @property
+    def phase(self) -> int:
+        """int: The phase of the Pauli string.
+
+        Can be one of +/-1
+        """
+        return self._phase
+
+    @property
+    def qubit_count(self) -> int:
+        """int: The number of qubits this Pauli string acts on."""
+        return self._qubit_count
+
+    def to_unsigned_observable(self) -> TensorProduct:
+        """Returns the observable corresponding to the unsigned part of the Pauli string.
+
+        For example, for a Pauli string -XYZ, the corresponding observable is X ⊗ Y ⊗ Z.
+
+        Returns:
+            TensorProduct: The tensor product of the unsigned factors in the Pauli string.
+        """
+
+        return TensorProduct(
+            [_PAULI_OBSERVABLES[self._nontrivial[qubit]] for qubit in sorted(self._nontrivial)]
+        )
+
+    def weight_n_substrings(self, weight: int) -> Tuple[PauliString, ...]:
+        r"""Returns every substring of this Pauli string with exactly `weight` nontrivial factors.
+
+        The number of substrings is equal to :math:`\binom{n}{w}`, where :math`n` is the number of
+        nontrivial (non-identity) factors in the Pauli string and :math`w` is `weight`.
+
+        Args:
+            weight (int): The number of non-identity factors in the substrings.
+
+        Returns:
+            Tuple[PauliString, ...]: A tuple of weight-n Pauli substrings.
+        """
+        substrings = []
+        for indices in itertools.combinations(self._nontrivial, weight):
+            factors = [
+                self._nontrivial[qubit]
+                if qubit in set(indices).intersection(self._nontrivial)
+                else "I"
+                for qubit in range(self._qubit_count)
+            ]
+            substrings.append(
+                PauliString(f"{PauliString._phase_to_str(self._phase)}{''.join(factors)}")
```

Comment:
It seems weird to have to do this, no? How about a factory method, or adding an optional phase parameter to the init? I'm not sure.

Reply:
I originally wrote the much more readable
```python
PauliString(f"{str(self)[0]}{''.join(factors)}")
```
but this just cuts out a bunch of unnecessary steps.

I don't think a separate phase is quite the right call when constructing from a string, but I can definitely envision a future constructor that takes in a phase and a list of factors, or Xs and Zs.